### ISSUE TITLE
[BUG] Azure Spring Apps deployment failed with Java 21

### ISSUE THREAD

**Issue by kirankumar994** [Azure/azure-sdk-for-java#41882]
**Describe the bug**
I have a spring boot application running with Java 17 which has been to deployed to azure spring apps and it is up and running. Recently, I have upgraded to Java 21 as well as Spring boot version to v3.3.3. The application is up and running on my local machine with latest versions. But, when I try to deploy it to azure spring apps, the deployment is getting failed with message says "InternalServerError". 

Later, I tried with simple spring demo application with Java 21. I have faced the same issue. Please find the below screenshots for the same. I am getting an option to select Java 21 runtime, but when I am trying to deploy with Java 21 it is failed. The azure spring apps documentation also says Java 21 is supported. I have created Azure spring apps with Standard plan pay as you go, East US region, FYI. Please help me out what is wrong here. Thanks in advance.

***Exception or Stack Trace***
![image](https://github.com/user-attachments/assets/cecc5d48-24ec-44a3-bf8e-1046ed89459d)

**To Reproduce**

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**
The deployemnt should be succeeded as it is shown for other applications with Java 17. 

**Screenshots**
![Screenshot from 2024-09-17 10-26-58](https://github.com/user-attachments/assets/66244f38-7fec-4786-912a-e7254b0af5dd)
![Screenshot from 2024-09-17 10-26-24](https://github.com/user-attachments/assets/2fb8e6d8-fb7b-4999-ba0d-4a486745fa3a)
![Screenshot from 2024-09-17 10-25-34](https://github.com/user-attachments/assets/e9733904-b4a1-401f-98a3-dceb0b283f77)
![Screenshot from 2024-09-17 10-24-42](https://github.com/user-attachments/assets/c35e38c9-6fcd-4fa8-ba7c-434c2ccc852f)

**Setup (please complete the following information):**
 
 - Java version: 21
 
 My local machine set up: 
 Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: /opt/maven
Java version: 21.0.4, vendor: Microsoft, runtime: /home/user/.sdkman/candidates/java/21.0.4-ms
Default locale: en_IN, platform encoding: UTF-8
OS name: "linux", version: "6.8.0-40-generic", arch: "amd64", family: "unix"

**Comment by saragluna** [comment]
@kirankumar994 Seems like not an issue related to the Spring Cloud Azure library but will ask the Azure Spring Apps folks to help take a look at it.

**Comment by kirankumar994** [comment]
Thank you for your reply, @saragluna . Could you please get it resolved as soon as possible as the new release of my application is on hold?

**Comment by saragluna** [comment]
@kirankumar994 the Azure Spring Apps team is working on it now.

**Comment by TobiasB91** [comment]
My company's application is blocked by this bug as well.

**Comment by saragluna** [comment]
Azure Spring Apps updates that the fix has been out. Could you please help check again.

**Comment by TobiasB91** [comment]
Thanks for the quick response! I tried to recreate my app this morning, but even though the InternalServerError is gone I'm not able to start my app. My AppEnvSpringAppConsoleLogs logs "Error occurred during initialization of VM" and "Too small maximum heap".

**Comment by kirankumar994** [comment]
Thanks for your reponse, @saragluna . I am able to deploy it with Java 21 only by uploading the jar file directly on the portal. 

![Screenshot from 2024-10-03 10-08-04](https://github.com/user-attachments/assets/5903b3ce-eab3-4cd7-87e4-87cb3ade1ebb)

But, however the application deployment is still failing from the pipelines. I am using the below plugin to deploy my application to the Azure spring apps. I have also created a separate app to double check the issue, but no luck though. 

![image](https://github.com/user-attachments/assets/efad514f-cfac-4a43-b955-9b2e25bbde00)


I am getting the below error when I try to deploy with the below cmd. 
"mvn azure-spring-apps:deploy"

![image (1)](https://github.com/user-attachments/assets/e433f7a9-bd36-4b2e-b63a-fb11dad8e498)

**Comment by kirankumar994** [comment]
Hi @saragluna , hope you had a chance to look at my comment. However, I managed to deploy my application with the below cmd in my pipelines. 

"az spring app deploy"

Seems like the issue lies on the plugin side where it doesn't have Java 21 runtime version not included, so it is taking Java 11 by default. Thanks for looking into it.

**Comment by saragluna** [comment]
Closing this issue now, please reopen it if there is further issue.
